src/jupyter_contrib_nbextensions/nbextensions/runtools/main.js

```diff
@@ -1,15 +1,153 @@
 // Add toolbar buttons for extended code execution commands
 
-
 define([
     'base/js/namespace',
     'jquery',
     'require',
     'base/js/events',
+    'services/config',
+    'base/js/utils',
     'codemirror/lib/codemirror'
-],   function(IPython, $, require, events, codemirror) {
+],   function(Jupyter, $, require, events, configmod, utils, CodeMirror) {
     "use strict";
 
+    var base_url = utils.get_body_data("baseUrl");
+    var config = new configmod.ConfigSection('notebook', {base_url: base_url});
+    var run_list = [];  /* list of cells to be run */
+
+    // define default config parameter values
+    var params = {
+        show_cellstate : false,
+        run_cells_above : 'Alt-a',
+        run_cells_below : 'Alt-b',
+        toggle_marker : 'Alt-t',
+        mark_all_codecells : 'Alt-m',
+        unmark_all_codecells : 'Alt-u',
+        run_marked_cells : 'Alt-r',
+        run_all_cells : 'Alt-x',
+        run_all_cells_ignore_errors : 'Alt-f',
+        stop_execution : 'Ctrl-c',
+        marked_color : '#0f0',
+        scheduled_color : '#00f',
+        run_color : '#f5f5f5'
+    };
+
+    // updates default params with any specified in the provided config data
+    var update_params = function (config_data) {
```

Comment:
this function never actually gets called. Also, given parameters are all `runtools.x`, it would make more sense to just do 

```javascript
$.extend(true, params, config_data.runtools);
```